rust/tool/src/os_release.rs

```diff
@@ -25,17 +25,36 @@ impl OsRelease {
         //
         // Because the ID field here does not have the same meaning as in a real os-release file,
         // it is fine to use a dummy value.
-        map.insert("ID", String::from("lanza"));
-        map.insert("PRETTY_NAME", generation.spec.bootspec.label.clone());
+        map.insert("ID".into(), String::from("lanza"));
+        map.insert("PRETTY_NAME".into(), generation.spec.bootspec.label.clone());
         map.insert(
-            "VERSION_ID",
+            "VERSION_ID".into(),
             generation
                 .describe()
                 .context("Failed to describe generation.")?,
         );
 
         Ok(Self(map))
     }
+
+    /// Parses the string representation of a os-release file.
+    pub fn from_str(value: &str) -> Result<Self> {
```

Comment:
This doesn't handle entries with quotes correctly. (See also the [spec](https://www.freedesktop.org/software/systemd/man/os-release.html).) They sort of work by accident, because you consider the quotes part of the value and then write them back out.

Maybe something like this would help: https://crates.io/crates/os-release

Reply:
I looked at the os-release crate because I had the same instinct of avoiding yolo parsing, but sadly it's (1) unmaintained since 2018 and (2) actually performs the same hack (just slightly less elegantly): Compare [mine](https://github.com/nix-community/lanzaboote/blob/smart-systemd-boot-install/rust/tool/src/os_release.rs#L53) with [theirs](https://github.com/pop-os/os-release/blob/master/src/lib.rs#L37)

We actually never write out a file we have previously read. So I'm not too worried about yolo parsing from that perspective. 
I should add a comment, however, that writing out a previously read file might not produce the same representation.

Reply:
Fair point. I've opened #77 to fix this later.